### PR TITLE
Feature/get soa endpoint

### PR DESCRIPTION
**Pull request by Joseph-Nash**

### CODE REVIEW

File: handlers/soa_resource.go [hunk 1]
```diff
@@ -111,3 +111,36 @@ func HandleCreateStatementOfAffairs(svc dao.Service) http.Handler {
...
+		log.InfoR(req, fmt.Sprintf("start GET request for get statement of affairs with transaction id: %s", transactionID))
+
+		statementOfAffairs, err := svc.GetStatementOfAffairsResource(transactionID)
+		if err != nil {
+			log.ErrorR(req, fmt.Errorf("failed to get statement of affairs from insolvency resource in db for transaction [%s]: %v", transactionID, err))
+			m := models.NewMessageResponse("there was a problem handling your request")
+			utils.WriteJSONWithStatus(w, req, m, http.StatusInternalServerError)
+			return
+		}
+		if statementOfAffairs.StatementDate == "" {
```
Comment (rsachdevch) [inline handlers/soa_resource.go:136]: Shouldn't we just be checking if the soa is object being returned is empty?

```suggestion
		if statementOfAffairs.StatementDate == (models.StatementOfAffairsResourceDao{}) {
```

File: handlers/soa_resource.go [hunk 2]
```diff
@@ -111,3 +111,36 @@ func HandleCreateStatementOfAffairs(svc dao.Service) http.Handler {
...
+			m := models.NewMessageResponse("statement of affairs not found on transaction")
```
Comment (rsachdevch) [inline handlers/soa_resource.go:137]: Should we pass in the transaction ID as well?

```suggestion
			m := models.NewMessageResponse(fmt.Sprintf("statement of affairs not found on transaction with ID: [%s]", transactionID))
```

File: handlers/soa_resource.go (hunk 1 above)
Comment (Joseph-Nash) [reply handlers/soa_resource.go:136]: This change wouldn't work, I believe because the struct has []string it cannot be compared with the == operator

Comment (rsachdevch) [reply handlers/soa_resource.go:136]: Okay that's fine then, we can leave the check as it is

**Review by rsachdevch — APPROVED** [review]

**Pull request merged**